### PR TITLE
genders: fix build on Linux

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
It looks like we need to tweak how this installs its Python plugin:
```
error: can't create or remove files in install directory

  The following error occurred while trying to add or remove files in the
  installation directory:
  
      [Errno 1] Operation not permitted: '/opt/homebrew/lib/python3.10/site-packages/test-easy-install-78591.write-test'
  
  The installation directory you specified (via --install-dir, --prefix, or
  the distutils default setting) was:
  
      /opt/homebrew/lib/python3.10/site-packages/
```

@carlocab I wonder if this was one of those packages that was secretly using Python 2 on macOS.  Interestingly the build works fine in Docker on Linux, so I'm not sure what's different in macOS.

**Comment by cho-m** [comment]
> ```
> error: can't create or remove files in install directory
> 
>   The following error occurred while trying to add or remove files in the
>   installation directory:
>   
>       [Errno 1] Operation not permitted: '/opt/homebrew/lib/python3.10/site-packages/test-easy-install-78591.write-test'
> ```

That looks like the Python 3.10 distutils prefix scheme issue where it tries to write outside sandbox.

Either can:
1. Manually set necessary install directories. Had to do this for #95399 and a few other formulae that are using deprecated distutils/setuptools installation.
2. Just disable Python bindings. Maybe `--with-python-extensions=no`

File: Formula/genders.rb [hunk 1]
```diff
@@ -22,13 +22,20 @@ class Genders < Formula
     sha256 cellar: :any, high_sierra:    "31a726904f22c156b763a8bc95bd3db6e85b8bc0cf7d8a82d584bb8684241f6c"
   end
 
+  depends_on "python@3.10" => :build
```
Comment (carlocab) [inline Formula/genders.rb:25]: Make this `uses_from_macos`

File: Formula/genders.rb [hunk 2]
```diff
@@ -22,13 +22,20 @@ class Genders < Formula
...
+  uses_from_macos "perl" => :build
+
   # Fix -flat_namespace being used on Big Sur and later.
   patch do
     url "https://raw.githubusercontent.com/Homebrew/formula-patches/03cf8088210822aa2c1ab544ed58ea04c897d9c4/libtool/configure-pre-0.4.2.418-big_sur.diff"
     sha256 "83af02f2aa2b746bb7225872cab29a253264be49db0ecebb12f841562d9a2923"
   end
 
   def install
+    ENV["PYTHON"] = Formula["python@3.10"].bin/"python3"
```
Comment (carlocab) [inline Formula/genders.rb:38]:
```suggestion
    ENV["PYTHON"] = which("python3")
```

File: Formula/genders.rb [hunk 3]
```diff
@@ -22,15 +22,27 @@ class Genders < Formula
...
   def install
+    ENV["PYTHON"] = which("python3")
     system "./configure", "--prefix=#{prefix}", "--with-java-extensions=no"
     system "make", "install"
+
+    # Move man page out of top level mandir on Linux
+    unless OS.mac?
+      mv prefix/"man/man3/Libgenders.3", man/"man3"
+      rm_rf prefix/"man"
+    end
```
Comment (carlocab) [inline Formula/genders.rb:45]:
```suggestion
    # Move man page out of top level mandir on Linux
    return if OS.mac?

    mv prefix/"man/man3/Libgenders.3", man/"man3"
    rm_rf prefix/"man"
```

File: Formula/genders.rb [hunk 4]
```diff
@@ -22,15 +22,27 @@ class Genders < Formula
...
   end
 
   def install
+    ENV["PYTHON"] = which("python3")
     system "./configure", "--prefix=#{prefix}", "--with-java-extensions=no"
     system "make", "install"
+
+    # Move man page out of top level mandir on Linux
+    unless OS.mac?
+      mv prefix/"man/man3/Libgenders.3", man/"man3"
```
Comment (carlocab) [inline Formula/genders.rb:43]: I don't quite follow what this does. It looks like it doesn't do anything?

Comment (danielnachun) [reply Formula/genders.rb:43]: I probably need a better comment here.  The issue is that on Linux the file `Libgenders.3` is incorrectly installed to `prefix/"man/man3"`.  This just moves the file out of that folder into `man/"man3"` and then deletes the top level `man` dir that `brew audit` doesn't allow.

Comment (carlocab) [reply Formula/genders.rb:43]: Ah yes, I forgot that `man` is in `share`. Doesn't setting this using `configure` flags work?

Comment (cho-m) [reply Formula/genders.rb:43]: Looks like common Perl manpage issue, where we often need to call `perl Makefile.PL ...` with `INSTALLSITEMAN3DIR` to override.

In this case, the Perl bindings are built internally during `make install`. The only args used are https://github.com/chaos/genders/blob/genders-1-27-3/src/extensions/perl/Libgenders/Makefile.am#L20-L27
```make
MAKEMAKERFLAGS = INSTALLDIRS=$(PERL_ARCH_INSTALL) \
		 DESTDIR=$(PERL_DESTDIR) \
		 PREFIX=$(prefix) \
		 INC=$(PERL_INC) \
		 LIBS=$(PERL_LIBS) \
		 GENDERS_LIBDIR=$(libdir) \
		 MAKEFILE=Makefile.xs \
		 FIRST_MAKEFILE=Makefile.xs
```

Comment (danielnachun) [reply Formula/genders.rb:43]: Is this something we should try to fix in the Makefile?  Or is this workaround acceptable (with a better comment)?  I can definitely confirm that setting `--mandir` with `configure` has no effect, probably for the reason @cho-m posted above.

File: Formula/genders.rb (hunk 3 above)
Comment (carlocab) [reply Formula/genders.rb:45]: Or, alternatively:

```suggestion
    # Move man page out of top level mandir on Linux
    share.install prefix/"man" unless OS.mac?
```

if that doesn't work:
```suggestion
    # Move man page out of top level mandir on Linux
    man.install (prefix/"man").children unless OS.mac?
```

File: Formula/genders.rb (hunk 4 above)
Comment (carlocab) [reply Formula/genders.rb:43]: Ok, that's fine. Let's just simplify this `unless` block. See below.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**